From 7ebb72a93922a0163a5b35c277f3bbd241bdf78c Mon Sep 17 00:00:00 2001
From: Yong Li <yong.b.li@linux.intel.com>
Date: Mon, 22 Oct 2018 16:20:36 +0800
Subject: [PATCH] Add shutdown policy interface for get/set shutdown policy OEM
 IPMI commands

The policy property is used to store the shutdown policy.

Tested by:
busctl get-property  "xyz.openbmc_project.Settings" \
"/xyz/openbmc_project/control/shutdown_policy_config" \
"xyz.openbmc_project.Control.ShutdownPolicy" "Policy"

busctl set-property  "xyz.openbmc_project.Settings" \
"/xyz/openbmc_project/control/shutdown_policy_config" \
"xyz.openbmc_project.Control.ShutdownPolicy" "Policy" y 1

Signed-off-by: Yong Li <yong.b.li@linux.intel.com>
---
 xyz/openbmc_project/Control/ShutdownPolicy.interface.yaml | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 xyz/openbmc_project/Control/ShutdownPolicy.interface.yaml

diff --git a/xyz/openbmc_project/Control/ShutdownPolicy.interface.yaml b/xyz/openbmc_project/Control/ShutdownPolicy.interface.yaml
new file mode 100644
index 0000000..e562ea8
--- /dev/null
+++ b/xyz/openbmc_project/Control/ShutdownPolicy.interface.yaml
@@ -0,0 +1,10 @@
+description: >
+    An interface for node shutdown policy on multi-node products.
+properties:
+    - name: Policy
+      type: byte
+      description: >
+          0: Do not shutdown node on a power supply over current(OC)
+          or a power supply over temperature(OT) event.
+          1: Shutdown node on an OC/OT event.
+          Only available on multi-node products.
-- 
2.7.4

